core/primitives/src/block_header.rs

```diff
@@ -357,13 +357,14 @@ impl BlockHeader {
         chunk_receipts_root: MerkleHash,
         chunk_headers_root: MerkleHash,
         chunk_tx_root: MerkleHash,
-        chunks_included: u64,
+        num_shards: u64,
         challenges_root: MerkleHash,
         timestamp: DateTime<Utc>,
         initial_gas_price: Balance,
         initial_total_supply: Balance,
         next_bp_hash: CryptoHash,
     ) -> Self {
+        let chunks_included = if height == 0 { num_shards } else { 0 };
```

Comment:
why does height matter here?

Reply:
old code was `Block::compute_chunks_included(&chunks, 0)` with `0` instead of `height`